src/Interpreters/Aggregator.cpp

```diff
@@ -1253,9 +1276,17 @@ void NO_INLINE Aggregator::executeImplBatch(
             /// Add only if the key already exists.
             auto find_result = state.findKey(method.data, i, *aggregates_pool);
             if (find_result.isFound())
+            {
                 aggregate_data = find_result.getMapped();
+            }
             else
+            {
+                /// If all keys are constant and this is new key
+                /// we don't need to do anything and just skip the whole block.
+                if (all_keys_are_const)
```

Comment:
What do you mean? Without this `if` we will execute aggregation on this block, but the result for the whole block will be saved in `overflow_row` (because all keys are the same), so, it will be actually unused. And we can avoid this extra work by early return here

Reply:
in my understanding it should be used for example if we have group by with totals and `totals_mode='before_having'`

Reply:
Ah, didn't know that, thanks! Will remove it